src/app/core.ts

```diff
@@ -1,9 +1,47 @@
 import {asperaDesktop} from '../index';
-import {client} from '../helpers/client';
-import {errorLog, generateErrorBody, generatePromiseObjects, getWebsocketUrl, isValidTransferSpec, randomUUID, throwError} from '../helpers/helpers';
 import {messages} from '../constants/messages';
+import Client from '../helpers/client';
+import {
+  errorLog,
+  generateErrorBody,
+  generatePromiseObjects,
+  getWebsocketUrl,
+  isSafari,
+  isValidTransferSpec,
+  randomUUID,
+  throwError
+} from '../helpers/helpers';
+import SafariExtensionHandler from '../helpers/safari-extension';
 import {DesktopInfo, TransferResponse} from '../models/aspera-desktop.model';
-import {DataTransferResponse, DesktopSpec, DesktopStyleFile, DesktopTransfer, FileDialogOptions, FolderDialogOptions, ModifyTransferOptions, ResumeTransferOptions, TransferSpec, CustomBrandingOptions} from '../models/models';
+import {
+  CustomBrandingOptions,
+  DataTransferResponse,
+  DesktopSpec,
+  DesktopStyleFile,
+  DesktopTransfer,
+  FileDialogOptions,
+  FolderDialogOptions,
+  ModifyTransferOptions,
+  ResumeTransferOptions,
+  TransferSpec
+} from '../models/models';
+
+let client: Client | undefined;
+let safariExtensionHandler: SafariExtensionHandler | undefined;
+
+if (!isSafari()) {
+  client = new Client();
+} else {
+  safariExtensionHandler = new SafariExtensionHandler();
+}
+
+function dispatchRequest(method: string, payload?: any): Promise<any> {
+  if (safariExtensionHandler) {
+    return safariExtensionHandler.sendRPCRequest(method, payload);
+  }
+
+  return client.request(method, payload);
+}
```

Comment:
Could we have the http client and safari client implement the same interface? Then we can keep `client.request` throughout and just rely on the conditional above to assign the right one at runtime.

It seems like both the safari and http clients should just expose a `request` method of some sort that returns a Promise.